waiter/integration/waiter/kubernetes_scheduler_integration_test.clj

```diff
@@ -194,12 +192,20 @@
       (let [current-user (retrieve-username)
             default-namespace (-> waiter-url default-scheduler-settings :replicaset-spec-builder :default-namespace)]
         (testing "Default namespace"
-          (check-pod-namespace waiter-url nil default-namespace))
+          (check-pod-namespace waiter-url {} default-namespace))
         (testing "Custom namespace"
-          (check-pod-namespace waiter-url current-user current-user))
-        (testing "Invalid namespace"
+          (check-pod-namespace waiter-url {:x-waiter-namespace current-user} current-user))
+        (testing "Run-as-requestor and namespace *"
+          (check-pod-namespace waiter-url {:x-waiter-namespace "*" :x-waiter-run-as-user "*"} current-user))
+        (testing "Invalid namespace for implicit current user"
           (is (thrown? Exception #"Service namespace must either be omitted or match the run-as-user"
-                       (check-pod-namespace waiter-url "not-current-user" current-user))))))))
+                       (check-pod-namespace waiter-url {:x-waiter-namespace "not-current-user"} current-user))))
+        (testing "Invalid namespace for run-as-requestor"
+          (is (thrown? Exception #"Cannot use run-as-requestor with a specific namespace"
+                       (check-pod-namespace waiter-url {:x-waiter-namespace "not-current-user" :x-waiter-run-as-user "*"} current-user))))
```

Comment:
Missing the case where the namespace is the same as the current running user.

Reply:
All 12 cases are covered now.